libglusterfs/src/common-utils.c

```diff
@@ -3063,6 +3063,12 @@ gf_set_volfile_server_common(cmd_args_t *cmd_args, const char *host,
     }
 
     char *lastptr = rindex(duphost, ':');
+
+    /* Handle IPV6 IP addr without port number */
+    if (lastptr && *(lastptr-1) == ':') {
```

Comment:
just a curious question. what happens if I pass ':' as the option here.

Reply:
When host is just passed as ":" , server->port would be assigned with default port 24007 as defined in port.
https://github.com/gluster/glusterfs/blob/devel/libglusterfs/src/common-utils.c#L3069. 
When *(lastptr - 1) was accessed even though no character was present at that index, no errors were reported.

Reply:
@mohit84 I agree. But for just fixing the IPv6 issue right now, this does the fix. 

@vatsa287 can you open an issue for regex parsing of the complete argument, and doing the necessary checks as a new issue ? We shall work on this separately.